pkg/api/meta/meta.go

```diff
@@ -68,14 +68,14 @@ func ListAccessor(obj interface{}) (List, error) {
 // required fields are missing. Fields that are not required return the default
 // value and are a no-op if set.
 func Accessor(obj interface{}) (Object, error) {
-	if oi, ok := obj.(ObjectMetaAccessor); ok {
-		if om := oi.GetObjectMeta(); om != nil {
+	if objectMetaAccessor, ok := obj.(ObjectMetaAccessor); ok {
+		if om := objectMetaAccessor.GetObjectMeta(); om != nil {
 			return om, nil
 		}
 	}
 	// we may get passed an object that is directly portable to Object
-	if oi, ok := obj.(Object); ok {
-		return oi, nil
+	if object, ok := obj.(Object); ok {
```

Comment:
I replied to you in the old commits but I it's lost after rebase. 
Anyway, In case both ObjectAccessor and Object are implemented, I think return objectMetaAccessor.GetObjectMeta is more favorable, because that only returns the ObjectMeta, rather than the entire object.